File: src/main/java/com/carpool/carpool/entity/register/RegisterUserRequest.java
package com.carpool.carpool.entity.register;

import jakarta.validation.constraints.NotBlank;

public class RegisterUserRequest {
    @NotBlank(message = "full name should not be blank")
    private String fullName;
    @NotBlank(message = "email should not be blank")
    private String email;
    @NotBlank(message = "password should not be blank")
    private String password;

    public RegisterUserRequest(String fullName, String email, String password) {
        this.fullName = fullName;
        this.email = email;
        this.password = password;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getEmail() {
        return email;
    }

    public String getFullName() {
        return fullName;
    }

    @Override
    public String toString() {
        return "RegisterUserRequest{" +
                "fullName='" + fullName + '\'' +
                ", email='" + email + '\'' +
                ", password='" + password + '\'' +
                '}';
    }
}

File: src/main/java/com/carpool/carpool/entity/register/RegisterUserRequest.kt
package com.carpool.carpool.entity.register

import jakarta.validation.constraints.NotBlank
import lombok.AllArgsConstructor
import lombok.Getter
import lombok.Setter
import lombok.ToString

@ToString
@Getter
@Setter
@AllArgsConstructor
data class RegisterUserRequest(
    val fullName: @NotBlank(message = "full name should not be blank") String? = null,
    val email: @NotBlank(message = "email should not be blank") String? = null,
    val password: @NotBlank(message = "password should not be blank") String? = null,
)